### ISSUE TITLE
Data formatting broken on datatable

### ISSUE THREAD

**Issue by ud3sh** [evidence-dev/evidence#617]
**Bug Description**

**Severity**
- [ ] Medium - I'm able to use Evidence, but a feature I need to use is not working

**Steps To Reproduce**
Let us know the steps we can take to reproduce the issue you're seeing. Sample data helps speed up the process of reproducing.
1. Create a MD file as follows
```
<DataTable data={income_vs_rent}/>

```income_vs_rent
select 'Location A' as geo_location, 371 as median_rent_usd, 12279 as income_per_capita_usd  union all  
select 'Location B' as geo_location, 1434 as median_rent_usd, 33882 as income_per_capita_usd  union all
select 'Location C' as geo_location, 698 as median_rent_usd, 26386 as income_per_capita_usd  union all
ORDER BY geo_location ASC```

```

Expected: First row DataTable shows income_per_capita as `$12.0k` 
Actual:  First row DataTable shows income_per_capita as `$12,279`

Note that the formatting shows up correctly on the query viewer table.

**Comment by hughess** [comment]
It turns out I missed adding the `columnUnitSummary` to the `formatValue` call in `DataTable`. Should be a straightforward fix, but will double check to make sure that adding it back in doesn't negatively affect any non-numeric types or the search functionality.

**Comment by hughess** [comment]
Fixed by #624
